shell/platform/embedder/embedder.h

```diff
@@ -25,6 +25,9 @@
 // - Function signatures (names, argument counts, argument order, and argument
 //   type) cannot change.
 // - The core behavior of existing functions cannot change.
+// - Instead of array of structures, prefer array of pointers to structures.
+//   This ensures that array indexing does not break if members are added
+//   to the structure.
```

Comment:
This LGTM.

As you noted separately, we could limit this to embedder-facing arrays of structs passed back in e.g. callbacks since for incoming structs, since for structs going the other way (being passed into the engine) we always use safe dereference macros, but we can't guarantee that third-party embedders do the same. 

All that said, I think there's a lot of value in keeping the advice here as succinct as possible and there are no downsides to doing this in both directions aside from making things a very small amount more onerous for embedders. @chinmaygarde probably has thoughts on this.